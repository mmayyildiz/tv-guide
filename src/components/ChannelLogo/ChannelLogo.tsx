import React from 'react';
import { BASE_URL, GET_CHANNEL_IMAGE_URL } from '../../constants/serviceUrlConstants';

interface ChannelLogoProps {
    sid: string;
    size: string;
}

const ChannelLogo = ({ sid, size }: ChannelLogoProps) => {
    return (
        <img src={`${BASE_URL}${GET_CHANNEL_IMAGE_URL}/${sid}/${size}.png`} alt="1" />
    )
}

export default ChannelLogo;