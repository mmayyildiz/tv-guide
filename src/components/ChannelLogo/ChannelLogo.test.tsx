import React from 'react';
import { shallow } from 'enzyme';
import ChannelLogo from './ChannelLogo';

const props = {
    sid: 'foo',
    size: '10'
};


const mountedComponent = shallow(<ChannelLogo {...props} />);

describe('ChannelLogo', () => {

    it('should render with props', () => {
        expect(mountedComponent).toMatchSnapshot();
    });

    it('should render a logo ', () => {
        expect(mountedComponent.find("img").prop("src")).toEqual("https://cdn.skyq.sky.com/recruitment/tvguide/logos/foo/10.png");

    });


});


