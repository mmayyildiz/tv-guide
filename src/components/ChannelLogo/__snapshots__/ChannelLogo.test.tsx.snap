// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelLogo should render with props 1`] = `ShallowWrapper {}`;
